modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/PathInterpreter.java

```diff
@@ -83,28 +84,23 @@ private String _getConfigurationEntryUUID(Matcher matcher) {
 		return matcher.group(3);
 	}
 
-	private FileVersion _getFileVersion(FileEntry fileEntry, long fileVersionId)
+	private FileVersion _getFileVersion(Matcher matcher, FileEntry fileEntry)
 		throws PortalException {
 
-		if (fileVersionId == 0) {
-			return fileEntry.getFileVersion();
-		}
-
-		return _dlAppService.getFileVersion(fileVersionId);
-	}
-
-	private long _getFileVersionId(Matcher matcher) {
 		if (matcher.group(2) == null) {
-			return 0;
+			List<FileVersion> fileVersions = fileEntry.getFileVersions(
+				WorkflowConstants.STATUS_APPROVED);
+
+			return fileVersions.get(0);
 		}
 
-		return Long.valueOf(matcher.group(2));
+		return fileEntry.getFileVersion(matcher.group(2));
```

Comment:
this method is different than _dlAppService.getFileVersion(fileVersionId);

fileEntry.getFileVersion(Long.valueOf(matcher.group(2)));

Reply:
Reverted to the old method, thank you!